### PR TITLE
update demo

### PR DESCRIPTION
**Pull request by miremond**
update to pac4j 1.7.0-SNAPSHOT
update to play-pac4j 1.4.0-SNAPSHOT
remove Google OpenID
Add Google OpenID Connect
Add stateless resource

### CODE REVIEW

File: app/controllers/Application.java [hunk 1]
```diff
@@ -91,4 +91,9 @@ public static Result theForm() throws TechnicalException {
         final FormClient formClient = (FormClient) Config.getClients().findClient("FormClient");
         return ok(views.html.theForm.render(formClient.getCallbackUrl()));
     }
+
+    @RequiresAuthentication(clientName = "BasicAuthClient", stateless = true)
```
Comment (leleuj) [inline app/controllers/Application.java]: Oh! I like this syntax: ```, stateless = true```.

File: app/controllers/Application.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public static Result samlIndex() {
         return protectedIndex();
     }
 
-    @RequiresAuthentication(clientName = "GoogleOpenIdClient")
+    @RequiresAuthentication(clientName = "OidcClient")
     public static Result googleOpenIdIndex() {
```
Comment (leleuj) [inline app/controllers/Application.java]: This mapping should be change by a ```oidcIndex()``` and the appropriate route.

File: app/controllers/Application.java (hunk 1 above)
Comment (leleuj) [inline app/controllers/Application.java]: But we use the same client for statefull and stateless flows?

File: app/controllers/Application.java [hunk 3]
```diff
@@ -91,4 +91,9 @@ public static Result theForm() throws TechnicalException {
         final FormClient formClient = (FormClient) Config.getClients().findClient("FormClient");
         return ok(views.html.theForm.render(formClient.getCallbackUrl()));
     }
+
+    @RequiresAuthentication(clientName = "BasicAuthClient", stateless = true)
+    public static Result statelessIndex() {
```
Comment (leleuj) [inline app/controllers/Application.java]: Where's the mapping for that in the ```routes``` file?

File: app/views/index.scala.html [hunk 4]
```diff
@@ -23,10 +22,7 @@
 <a href="@urlBasicAuth">Authenticate with basic auth</a><br />
 <a href="@urlCas">Authenticate with CAS</a><br />
 <a href="@urlSaml">Authenticate with SAML</a><br />
-<form action="@urlGoogleOpenId" method="POST">
-  <input type="text" name="openIdUser" value="https://www.google.com/accounts/o8/id" />
-  <input type="submit" value="Authenticate with Google OpenID" />
-</form>
+<a href="@urlGoogleOpenId">Authenticate with OpenID Connect</a><br />
```
Comment (leleuj) [inline app/views/index.scala.html]: I was expecting ```@urlOidc```...

File: app/controllers/Application.java (hunk 1 above)
Comment (miremond) [inline app/controllers/Application.java]: yes, this is the promise of pac4j ;)

**Pull request merged**